ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/Basics.hs

```diff
@@ -208,7 +208,8 @@ distribTopLevelConfig ei tlc =
              (completeLedgerConfig'    ei cfgLedger)
              cfgBlock
              cfgCodec
-             cfgStorage))
+             cfgStorage
+             emptyCheckpointsMap))
```

Comment:
As mentioned in the last paragraph in https://github.com/IntersectMBO/ouroboros-consensus/pull/453#issuecomment-1903909527 (and in the subsequent replies), I think this is fine, but I think it makes sense to

 - either document in a comment why it is fine to give an empty `CheckpointsMap` to the per-era `TopLevelConfig`s even if the one in the provided `tlc :: TopLevelConfig (HardForkBlock xs)` is non-empty, 
 - or convert the `CheckpointsMap` in `tlc` to each era as done in #453 via `ConvertRawHash`:
   https://github.com/IntersectMBO/ouroboros-consensus/blob/eb1740fc03ad5b565c69e5053bf9a38333e9fd1f/ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/Node.hs#L778-L779
   (needs a bit of SOP fun)

Reply:
Added a comment in 9928f734.

> or convert the CheckpointsMap in tlc to each era

While this is an interesting idea, I think it would deserve a change of types.

`CheckpointsMap blk` hints too strongly at the fact that it describes checkpoints of blocks `blk`. If we were to coerce the maps as in #453, we would end up with maps that contain hashes potentially referring to blocks of various types.

Likely we would need to switch to using `ShortByteString` instead of `HeaderHash blk`. If we then switch the types, we don't need a coercion anymore. Let me know if you want to try this.

Reply:
FTR: I think the status quo with the comment is fine.